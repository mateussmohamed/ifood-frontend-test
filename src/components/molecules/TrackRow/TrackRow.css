.track-row {
  list-style: none;
  color: #7380af;
  padding: 10px 0;
  border-bottom: 1px solid #1f2848;
  transition: all 0.2s;
  display: flex;
  cursor: pointer;
}

.track-row:hover {
  color: #ffffff;
}

.track-row:hover .play-button {
  opacity: 1;
}

.play-button {
  cursor: pointer;
  width: 20px;
  height: 20px;
  border: none;
  margin-right: 10px;
  opacity: 0;
  outline: none;
  background-size: 100%;
  background-color: transparent;
  background-image: url('/images/play-button.png');
}

@media screen and (max-width: 768px) {
  .track-row .artist {
    display: none;
  }
}

@media screen and (max-width: 480px) {
  .track-row .album {
    display: none;
  }
}
