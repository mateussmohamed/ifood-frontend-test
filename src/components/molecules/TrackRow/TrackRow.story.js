import { storiesOf } from '@storybook/react';
import React from 'react';

import TrackRow from './index';

const stories = storiesOf('TrackRow', module);

const labels = {
  title: 'Congratulations',
  artist: 'Post Malone',
  album: 'Stoney',
  duration: '3:40'
};

stories.add('default', () => (
  <section>
    <TrackRow labels={labels} />
  </section>
));
