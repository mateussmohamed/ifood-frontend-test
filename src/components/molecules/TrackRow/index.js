import React from 'react';
import PropTypes from 'prop-types';
import TrackLabel from '../../atoms/TrackLabel';
import './TrackRow.css';

const TrackRow = ({ labels, action }) => (
  <li className="track-row" onClick={action}>
    <button className="play-button"></button>
    {Object.keys(labels).map(key => (
      <TrackLabel
        type={key}
        text={labels[key]}
        key={`track-label-${key}`}
      />
    ))}
  </li>
);

TrackRow.defaultProps = {
  labels: {},
};

TrackRow.propTypes = {
  labels: PropTypes.objectOf(PropTypes.any),
  action: PropTypes.func.isRequired,
};

export default TrackRow;
