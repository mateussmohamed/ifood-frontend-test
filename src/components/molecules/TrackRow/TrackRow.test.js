import React from 'react';
import renderer from 'react-test-renderer';
import TrackRow from './index';

it('Should TrackRow match to snapshot', () => {
  const labels = {
    title: 'Congratulations',
    artist: 'Post Malone',
    album: 'Stoney',
    duration: '3:40'
  };

  const component = <TrackRow labels={labels} />;
  const tree = renderer.create(component).toJSON();
  expect(tree).toMatchSnapshot();
});
