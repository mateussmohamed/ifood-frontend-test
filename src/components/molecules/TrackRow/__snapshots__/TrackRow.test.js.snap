// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should TrackRow match to snapshot 1`] = `
<li
  className="track-row"
  onClick={[Function]}
>
  <button
    className="play-button"
  />
  <span
    className="track-label title"
  >
    Congratulations
  </span>
  <span
    className="track-label artist"
  >
    Post Malone
  </span>
  <span
    className="track-label album"
  >
    Stoney
  </span>
  <span
    className="track-label duration"
  >
    3:40
  </span>
</li>
`;
